aw/ViewControllers/Reaches/Detail/RunDetailTableViewController.swift

```diff
@@ -0,0 +1,234 @@
+import CoreData
+import UIKit
+
+class RunDetailTableViewController: UITableViewController {
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var sectionLabel: UILabel!
+    @IBOutlet weak var difficultyLabel: UILabel!
+    @IBOutlet weak var lengthLabel: UILabel!
+    @IBOutlet weak var gradientLabel: UILabel!
+    @IBOutlet weak var readingLabel: UILabel!
+    @IBOutlet weak var unitsLabel: UILabel!
+    @IBOutlet weak var detailUpdated: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var runnabilityLabel: UILabel!
+
+    var managedObjectContext: NSManagedObjectContext?
+
+    var reach: Reach?
+
+    var expandDescription: Bool = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        initialize()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if let reach = reach,
+            let context = managedObjectContext,
+            reach.detailUpdated == nil {
+            drawView()
+            print("Updating reach detail")
+            AWApiHelper.updateReachDetail(reachID: String(reach.id), viewContext: context) {
+                print("Updated reach details")
+                self.drawView()
+            }
+            AWApiHelper.updateReaches(reachIds: [String(reach.id)], viewContext: context) {
+                self.drawView()
+            }
+        } else {
+            print("Details already updated")
+            drawView()
+        }
+    }
+
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 4
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0, 2:
```

Comment:
You should at least document what these numbers refer to, I have no idea from reading the code.